File: java/src/localsearchoptimization/examples/structures/permutation/Swap.java
package localsearchoptimization.examples.structures.permutation;

import localsearchoptimization.components.Configuration;
import localsearchoptimization.components.Operator;
import localsearchoptimization.components.Solution;
import localsearchoptimization.examples.structures.TwoOperands;

public class Swap extends Operator {

    public Swap(int elementsNumber) {
        this(elementsNumber, 1);
    }

    public Swap(int elementsNumber, double weight) {
        super(weight);
        for (int i = 0; i < elementsNumber - 1; i++)
            for (int j = i + 1; j < elementsNumber; j++)
                configurations.add(new TwoOperands(i, j, this));
    }

    @Override
    public Solution apply(Solution solution, Configuration configuration) {
        Permutation permutation = (Permutation) solution;
        TwoOperands operands = (TwoOperands) configuration;

        int[] swapped = permutation.getPermutation().clone();

        swapped[operands.first] = permutation.getPermutation()[operands.second];
        swapped[operands.second] = permutation.getPermutation()[operands.first];

        return permutation.fetchPermutation(swapped, "swap");
    }
}

File: C#/LocalSearchOptimization/Examples/Structures/Permutation/Swap.cs
using System.Collections.Generic;
using LocalSearchOptimization.Components;

namespace LocalSearchOptimization.Examples.Structures.Permutation
{
    public class Swap : Operator
    {
        public Swap(int elementsNumber, double weight = 1) : base(weight)
        {
            for (int i = 0; i < elementsNumber - 1; i++)
                for (int j = i + 1; j < elementsNumber; j++)
                    Configurations.Add(new TwoOperands(i, j, this));
        }

        public override ISolution Apply(ISolution solution, Configuration configuration)
        {
            IPermutation permutation = (IPermutation)solution;
            TwoOperands operands = (TwoOperands)configuration;

            List<int> swapped = new List<int>(permutation.Order)
            {
                [operands.First] = permutation.Order[operands.Second],
                [operands.Second] = permutation.Order[operands.First]
            };

            return permutation.FetchPermutation(swapped, "swap");
        }
    }
}